### PR TITLE
[PM-14542] Move kdf to km ownership

### PR DESCRIPTION
**Pull request by quexten**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-14542

## 📔 Objective

Move KDF config service and config model to km ownership.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: libs/key-management/src/models/kdf-config.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { Jsonify } from "type-fest";
 
-import { KdfType } from "../../../platform/enums/kdf-type.enum";
-import { RangeWithDefault } from "../../../platform/misc/range-with-default";
+import { KdfType } from "../../../common/src/platform/enums/kdf-type.enum";
```
Comment (mzieniukbw) [inline libs/key-management/src/models/kdf-config.ts:3]: ❓  Shouldn't we own `KdfType` too ?

File: libs/common/src/auth/services/key-connector.service.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { firstValueFrom } from "rxjs";
 
 import { LogoutReason } from "@bitwarden/auth/common";
+import { Argon2KdfConfig, KdfConfig, PBKDF2KdfConfig } from "@bitwarden/key-management";
 
 import { KeyService } from "../../../../key-management/src/abstractions/key.service";
```
Comment (mzieniukbw) [inline libs/common/src/auth/services/key-connector.service.ts:6]: Use `@bitwarden/key-management`

File: libs/common/src/auth/services/user-verification/user-verification.service.spec.ts [hunk 3]
```diff
@@ -7,7 +7,9 @@ import {
   UserDecryptionOptions,
   UserDecryptionOptionsServiceAbstraction,
 } from "@bitwarden/auth/common";
+import { KdfConfig } from "@bitwarden/key-management";
 
+import { KdfConfigService } from "../../../../../key-management/src/abstractions/kdf-config.service";
 import { KeyService } from "../../../../../key-management/src/abstractions/key.service";
```
Comment (mzieniukbw) [inline libs/common/src/auth/services/user-verification/user-verification.service.spec.ts:13]: Use `@bitwarden/key-management`

File: libs/common/src/auth/services/user-verification/user-verification.service.ts [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import { firstValueFrom, map } from "rxjs";
 
 import { UserDecryptionOptionsServiceAbstraction } from "@bitwarden/auth/common";
+import { KdfConfigService } from "@bitwarden/key-management";
 
 import { PinServiceAbstraction } from "../../../../../auth/src/common/abstractions/pin.service.abstraction";
 import { KeyService } from "../../../../../key-management/src/abstractions/key.service";
```
Comment (mzieniukbw) [inline libs/common/src/auth/services/user-verification/user-verification.service.ts:7]: Use `@bitwarden/key-management`

File: libs/common/src/tools/send/services/send.service.ts [hunk 5]
```diff
@@ -1,7 +1,8 @@
 import { Observable, concatMap, distinctUntilChanged, firstValueFrom, map } from "rxjs";
 
+import { PBKDF2KdfConfig } from "@bitwarden/key-management";
+
 import { KeyService } from "../../../../../key-management/src/abstractions/key.service";
```
Comment (mzieniukbw) [inline libs/common/src/tools/send/services/send.service.ts:5]: Use `@bitwarden/key-management`

File: libs/common/tsconfig.json [hunk 6]
```diff
@@ -6,7 +6,10 @@
     "./custom-matchers.d.ts",
     "../key-management/src/key.service.spec.ts",
     "../key-management/src/key.service.ts",
-    "../key-management/src/abstractions/key.service.ts"
+    "../key-management/src/abstractions/key.service.ts",
+    "../key-management/src/abstractions/kdf-config.service.ts",
+    "../key-management/src/kdf-config.service.spec.ts",
+    "../key-management/src/kdf-config.service.ts"
```
Comment (mzieniukbw) [inline libs/common/tsconfig.json:12]: ❓ Can't we just do this ? `"../key-management/src/index.ts"`

**Review by mzieniukbw — CHANGES_REQUESTED** [review]

**Review by jrmccannon — APPROVED** [review]
Changes to admin console code are good.

**Review by djsmith85 — COMMENTED** [review]
Tools changes look good, besides the one thing @mzieniukbw already commented on

File: apps/desktop/src/app/services/services.module.ts [hunk 7]
```diff
@@ -84,6 +80,8 @@ import { CipherService as CipherServiceAbstraction } from "@bitwarden/common/vau
 import { DialogService, ToastService } from "@bitwarden/components";
 import { PasswordGenerationServiceAbstraction } from "@bitwarden/generator-legacy";
 import {
+  KdfConfigService,
+  KdfConfigService as KdfConfigServiceAbstraction,
```
Comment (mzieniukbw) [inline apps/desktop/src/app/services/services.module.ts:84]: ❓ Shouldn't this be instead:
```
DefaultKdfConfigService,
KdfConfigService
```

**Review by mzieniukbw — APPROVED** [review]

**Review by djsmith85 — APPROVED** [review]
Approving for tools

**Review by Jingo88 — APPROVED** [review]

**Review by JaredSnider-Bitwarden — APPROVED** [review]

**Review by jrmccannon — APPROVED** [review]

**Pull request merged**